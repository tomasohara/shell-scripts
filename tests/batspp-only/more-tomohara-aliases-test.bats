#!/usr/bin/env bats


# Make executables ./tests/../ visible to PATH
PATH="/home/aveey/tom-project/shell-scripts/tests/batspp-only/../:$PATH"

# Enable aliases
shopt -s expand_aliases



@test "test-1" {
	testfolder="/tmp/batspp-95327/test-1"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'basename $PWD\n' "=========="
	test-1-actual 
	echo "=========" $'tests' "========="
	test-1-expected 
	echo "============================"
	# ???: 'basename $PWD\n'=$(test-1-actual)
	# ???: 'tests'=$(test-1-expected)
	[ "$(test-1-actual)" == "$(test-1-expected)" ]
}

function test-1-actual () {
	# no-op in case content just a comment
	true

	basename $PWD

}

function test-1-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
tests
END_EXPECTED
}


@test "test-3" {
	test-3-setup;
	echo "==========" $'/bin/rm -rf "/tmp/test-dir-15485863"\n' "=========="
	test-3-actual 
	echo "=========" $'$ test_dir="/tmp/test-dir-15485863"\n$ mkdir -p "$test_dir"\n$ command cd "$test_dir"' "========="
	test-3-expected 
	echo "============================"
	# ???: '/bin/rm -rf "/tmp/test-dir-15485863"\n'=$(test-3-actual)
	# ???: '$ test_dir="/tmp/test-dir-15485863"\n$ mkdir -p "$test_dir"\n$ command cd "$test_dir"'=$(test-3-expected)
	[ "$(test-3-actual)" == "$(test-3-expected)" ]
}

function test-3-actual () {
	# no-op in case content just a comment
	true

	/bin/rm -rf "/tmp/test-dir-15485863"

}

function test-3-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ test_dir="/tmp/test-dir-15485863"
$ mkdir -p "$test_dir"
$ command cd "$test_dir"
END_EXPECTED
}

function test-3-setup () {
	# no-op in case content just a comment
	true

	testfolder="/tmp/batspp-95327/test-3"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"


}


@test "test-4" {
	testfolder="/tmp/batspp-95327/test-4"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'ls ../*.html\n' "=========="
	test-4-actual 
	echo "=========" $'../template.html' "========="
	test-4-expected 
	echo "============================"
	# ???: 'ls ../*.html\n'=$(test-4-actual)
	# ???: '../template.html'=$(test-4-expected)
	[ "$(test-4-actual)" == "$(test-4-expected)" ]
}

function test-4-actual () {
	# no-op in case content just a comment
	true

	ls ../*.html

}

function test-4-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
../template.html
END_EXPECTED
}


@test "test-5" {
	testfolder="/tmp/batspp-95327/test-5"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'dir=$(realpath "..")\n' "=========="
	test-5-actual 
	echo "=========" $'$ url-path ../template.html | perl -pe "s@$dir@/x/y/z@;"\nfile:////x/y/z/template.html' "========="
	test-5-expected 
	echo "============================"
	# ???: 'dir=$(realpath "..")\n'=$(test-5-actual)
	# ???: '$ url-path ../template.html | perl -pe "s@$dir@/x/y/z@;"\nfile:////x/y/z/template.html'=$(test-5-expected)
	[ "$(test-5-actual)" == "$(test-5-expected)" ]
}

function test-5-actual () {
	# no-op in case content just a comment
	true

	dir=$(realpath "..")

}

function test-5-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ url-path ../template.html | perl -pe "s@$dir@/x/y/z@;"
file:////x/y/z/template.html
END_EXPECTED
}


@test "test-6" {
	testfolder="/tmp/batspp-95327/test-6"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'umask ug=rwx,o=r\n' "=========="
	test-6-actual 
	echo "=========" $'$ mkdir abc def\n$ touch ghi jkl\n$ echo > mno\n$ dir | perl -pe "s/ (\\d+) [\\w ]{2,8}/ \\1 username /; s/\\w\\w\\w \\d+ \\d\\d:\\d\\d/mon \\1 hh:mm/;";' "========="
	test-6-expected 
	echo "============================"
	# ???: 'umask ug=rwx,o=r\n'=$(test-6-actual)
	# ???: '$ mkdir abc def\n$ touch ghi jkl\n$ echo > mno\n$ dir | perl -pe "s/ (\\d+) [\\w ]{2,8}/ \\1 username /; s/\\w\\w\\w \\d+ \\d\\d:\\d\\d/mon \\1 hh:mm/;";'=$(test-6-expected)
	[ "$(test-6-actual)" == "$(test-6-expected)" ]
}

function test-6-actual () {
	# no-op in case content just a comment
	true

	umask ug=rwx,o=r

}

function test-6-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ mkdir abc def
$ touch ghi jkl
$ echo > mno
$ dir | perl -pe "s/ (\d+) [\w ]{2,8}/ \1 username /; s/\w\w\w \d+ \d\d:\d\d/mon \1 hh:mm/;";
END_EXPECTED
}
