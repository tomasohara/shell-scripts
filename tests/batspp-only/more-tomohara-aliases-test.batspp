# More tests for tomohara-aliases.bash

## TEST: cd /usr/local/misc/programs/bash/shell-scripts-aviyan/tests

$ basename $PWD
tests

# Setup temp. dir
# note: variable not used in rm as a precaution
$ /bin/rm -rf "/tmp/test-dir-15485863"
$ test_dir="/tmp/test-dir-15485863"
$ mkdir -p "$test_dir"
$ command cd "$test_dir"

#===============================================================================
# TODO: find jupyter way to do divider line (or at least bash)

# test url-path

$ ls ../*.html
../template.html

$ dir=$(realpath "..")
$ url-path ../template.html | perl -pe "s@$dir@/x/y/z@;"
file:////x/y/z/template.html

# Test dir alias 
# note: time-dependent details changed
$ umask ug=rwx,o=r
$ mkdir abc def
$ touch ghi jkl
$ echo > mno
$ dir | perl -pe "s/ (\d+) [\w ]{2,8}/ \1 username /; s/\w\w\w \d+ \d\d:\d\d/mon \1 hh:mm/;";
