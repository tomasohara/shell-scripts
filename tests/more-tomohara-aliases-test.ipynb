{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# More tests for tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "## TEST: cd /usr/local/misc/programs/bash/shell-scripts-aviyan/tests\n",
    "## For number of digits not being critical\n",
    "alias testnum=\"sed -r \"s/[0-9][0-9]*/N/g\"\"\n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tests\n"
     ]
    }
   ],
   "source": [
    "basename $PWD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setup temp. dir\n",
    "# note: variable not used in rm as a precaution\n",
    "# BAD: /bin/rm -rf \"/tmp/test-dir-15485863\"\n",
    "\n",
    "## OLD: Used the concept of trash-dir\n",
    "# test_dir=\"/tmp/test-dir-15485863\"\n",
    "# trash_dir=$TMP/\"_temp-trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "# mkdir -p \"$test_dir\"\n",
    "# mkdir -p \"$trash_dir\"\n",
    "# command cd \"$test_dir\"\n",
    "\n",
    "\n",
    "TMP=${TMP:-/tmp}\n",
    "if [ \"$DEBUG_LEVEL\" -lt 4 ]; then TMP=/tmp/test-more-tomohara-aliases; fi\n",
    "temp_dir=$TMP/more-tomohara-aliases-test\n",
    "rename-with-file-date \"$temp_dir\"\n",
    "command mkdir -p \"$temp_dir\"\n",
    "command cd \"$temp_dir\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#===============================================================================\n",
    "# TODO: find jupyter way to do divider line (or at least bash)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# test url-path\n",
    "echo \"<html>This is a HTML file</html>\" > template.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "template.html\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# ls ../*.html\n",
    "command ls *.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "dir=$(realpath \"..\")\n",
    "## OLD: Seperated to next cell\n",
    "# url-path ../template.html | perl -pe \"s@$dir@/x/y/z@;\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "url-path ../template.html | perl -pe \"s@$dir@/x/y/z@;\" > url_path.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "cat url_path.out | grep -q \"file:////x/y/z/template.html\" && echo \"True\" || echo \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Test dir alias \n",
    "# # note: time-dependent details changed\n",
    "umask ug=rwx,o=r\n",
    "\n",
    "## OLD: Seperated to cells\n",
    "# mkdir abc def\n",
    "# touch ghi jkl\n",
    "# echo > mno\n",
    "# dir | perl -pe \"s/ (\\d+) [\\w ]{2,8}/ \\1 username /; s/\\w\\w\\w \\d+ \\d\\d:\\d\\d/mon \\1 hh:mm/;\";"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir abc def\n",
    "touch ghi jkl\n",
    "echo > mno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# dir | perl -pe \"s/ (\\d+) [\\w ]{2,8}/ \\1 username /; s/\\w\\w\\w \\d+ \\d\\d:\\d\\d/mon \\1 hh:mm/;\"\n",
    "dir | perl -pe \"s/ (\\d+) [\\w ]{2,8}/ \\1 username /; s/\\w\\w\\w \\d+ \\d\\d:\\d\\d/mon \\1 hh:mm/;\" > dir_perl.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw-r-- N username er    N mon  hh:mm mno\n"
     ]
    }
   ],
   "source": [
    "cat dir_perl.out | head -n 4 | tail -n 1 | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "END\n"
     ]
    }
   ],
   "source": [
    "# Remove everything after use\n",
    "## OLD: command mv -f * \"$trash_dir\"\n",
    "echo \"END\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
