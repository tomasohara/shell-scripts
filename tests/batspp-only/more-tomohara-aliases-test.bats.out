1..5
not ok 1 test-1
# (in test file batspp-only/more-tomohara-aliases-test.bats, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== basename $PWD
#  ==========
# test-1
# ========= tests =========
# tests
# ============================
not ok 2 test-3
# (in test file batspp-only/more-tomohara-aliases-test.bats, line 55)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== /bin/rm -rf "/tmp/test-dir-15485863"
#  ==========
# ========= $ test_dir="/tmp/test-dir-15485863"
# $ mkdir -p "$test_dir"
# $ command cd "$test_dir" =========
# $ test_dir="/tmp/test-dir-15485863"
# $ mkdir -p ""
# $ command cd ""
# ============================
not ok 3 test-4
# (from function `test-4-actual' in file batspp-only/more-tomohara-aliases-test.bats, line 110,
#  in test file batspp-only/more-tomohara-aliases-test.bats, line 97)
#   `test-4-actual ' failed with status 2
# ========== ls ../*.html
#  ==========
# ls: cannot access '../*.html': No such file or directory
not ok 4 test-5
# (in test file batspp-only/more-tomohara-aliases-test.bats, line 137)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== dir=$(realpath "..")
#  ==========
# ========= $ url-path ../template.html | perl -pe "s@$dir@/x/y/z@;"
# file:////x/y/z/template.html =========
# $ url-path ../template.html | perl -pe "s@/tmp/batspp-95327@/x/y/z@;"
# file:////x/y/z/template.html
# ============================
not ok 5 test-6
# (in test file batspp-only/more-tomohara-aliases-test.bats, line 172)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== umask ug=rwx,o=r
#  ==========
# ========= $ mkdir abc def
# $ touch ghi jkl
# $ echo > mno
# $ dir | perl -pe "s/ (\d+) [\w ]{2,8}/ \1 username /; s/\w\w\w \d+ \d\d:\d\d/mon \1 hh:mm/;"; =========
# $ mkdir abc def
# $ touch ghi jkl
# $ echo > mno
# $ dir | perl -pe "s/ (\d+) [\w ]{2,8}/ \1 username /; s/\w\w\w \d+ \d\d:\d\d/mon \1 hh:mm/;";
# ============================
